hedera-node/src/main/java/com/hedera/services/grpc/marshalling/FeeAssessor.java

```diff
@@ -78,20 +108,12 @@ public ResponseCodeEnum assess(
 					htsFeeAssessor.assess(payer, fee, balanceChangeManager, accumulator);
 				}
 				if (balanceChangeManager.numChangesSoFar() > maxBalanceChanges) {
-					return CUSTOM_FEE_CHARGING_EXCEEDED_MAX_ACCOUNT_AMOUNTS;
+					return FixedFeeProcessingResult.ASSESSMENT_FAILED_WITH_TOO_MANY_ADJUSTMENTS_REQUIRED;
 				}
 			} else {
-				hasFractionalFees = true;
+				return FixedFeeProcessingResult.FRACTIONAL_FEE_ASSESSMENT_PENDING;
```

Comment:
Oops, here we only want to "remember to return" `FRACTIONAL_FEE_ASSESSMENT_PENDING`.

 If we `return` right away, we will skip any other fixed fees in the `fees` list! 😲

Reply:
Right, right, that's silly!

Fixed and also added test cases to cover this scenario.